core/templates/dev/head/domain/collection/CollectionUpdateService.js

```diff
@@ -99,6 +99,28 @@ oppia.factory('CollectionUpdateService', [
         return _getParameterFromChangeDict(changeDict, 'exploration_id');
       };
 
+      // Validates the tags for the collection
+      var _validateTags = function(tags) {
+        // Check to ensure that all tags follow the format specified in
+        // TAG_REGEX.
+        for (var i = 0; i < tags.length; i++) {
+          var tagRegex = new RegExp(GLOBALS.TAG_REGEX);
+          if (!tags[i].match(tagRegex)) {
+            return false;
+          }
+        }
+
+        // Check to ensure that there are no duplicate tags.
+        for (var i = 0; i < tags.length; i++) {
```

Comment:
you could try `tags.every(function (tag, idx) { return tags.indexOf(tag, idx + 1) === -1; });`